### PR TITLE
[datadog_dashboard] Support Formula and Function requests in heatmap widget

### LINKED ISSUE #2096: Existing heatmap widget failed to pull in plan
### Datadog Terraform Provider Version

3.29.0

### Terraform Version

1.5.7

### What resources or data sources are affected?

datadog_dashboard

### Terraform Configuration Files

_No response_

### Relevant debug or panic output

_No response_

### Expected Behavior

When importing an existing dashboard with the following configuration:
```
{
    "viz": "heatmap",
    "events": [
        {
            "q": "Deployed to Production"
        }
    ],
    "requests": [
        {
            "style": {
                "palette": "dog_classic"
            },
            "response_format": "timeseries",
            "queries": [
                {
                    "data_source": "metrics",
                    "name": "query1",
                    "query": "max:system.load.1{$site,$az,$tg,aws:cloudformation:stack-name:...} by {host}"
                },
                {
                    "data_source": "metrics",
                    "name": "query2",
                    "query": "max:system.load.1{$site,$az,$tg,aws:cloudformation:stack-name:...} by {host}"
                }
            ],
            "formulas": [
                {
                    "formula": "query1"
                },
                {
                    "formula": "query2"
                }
            ]
        }
    ]
}
```
a `terraform plan` run should detect and show the correct existing values.

### Actual Behavior

Plan shows only this:
```
      - widget {
          - id = 107436315 -> null

          - heatmap_definition {
              - show_legend = false -> null
              - title       = "Max System Load by host" -> null

              - event {
                  - q = "Deployed to Production" -> null
                }

              - request {

                  - style {
                      - palette = "dog_classic" -> null
                    }
                }
            }
        }
```

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by nkzou**
Resolves #2096
Bumps go sdk and adds missing formula and function fields to heatmap widgets.

### CODE REVIEW

**Review by skarimo — APPROVED** [review]

**Comment by maycmlee** [comment]
Created [DOCS-6203](https://datadoghq.atlassian.net/browse/DOCS-6203) for docs review.

**Review by DrkSephy — DISMISSED** [review]

File: docs/resources/dashboard.md [hunk 1]
```diff
@@ -3798,6 +3853,173 @@ Optional:
...
+- `apm_resource_stats_query` (Block List, Max: 1) The APM Resource Stats query using formulas and functions. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--apm_resource_stats_query))
+- `event_query` (Block List, Max: 1) A timeseries formula and functions events query. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--event_query))
+- `metric_query` (Block List, Max: 1) A timeseries formula and functions metrics query. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--metric_query))
+- `process_query` (Block List, Max: 1) The process query using formulas and functions. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--process_query))
+- `slo_query` (Block List, Max: 1) The slo query using formulas and functions. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--slo_query))
```
Comment (estherk15) [inline docs/resources/dashboard.md:3866]:
```suggestion
- `slo_query` (Block List, Max: 1) The SLO query using formulas and functions. (see [below for nested schema](#nestedblock--widget--group_definition--widget--heatmap_definition--request--query--slo_query))
```

File: docs/resources/dashboard.md [hunk 2]
```diff
@@ -3798,6 +3853,173 @@ Optional:
...
+
+<a id="nestedblock--widget--group_definition--widget--heatmap_definition--request--query--apm_dependency_stats_query"></a>
+### Nested Schema for `widget.group_definition.widget.heatmap_definition.request.query.apm_dependency_stats_query`
+
+Required:
+
+- `data_source` (String) The data source for APM Dependency Stats queries. Valid values are `apm_dependency_stats`.
+- `env` (String) APM Environment.
```
Comment (estherk15) [inline docs/resources/dashboard.md:3874]: I wasn't sure about this, but ignore if this should be capitalized!
```suggestion
- `env` (String) APM environment.
```

File: docs/resources/dashboard.md [hunk 3]
```diff
@@ -9387,6 +9611,59 @@ Optional:
...
+- `metric` (String) The metric from the request to correlate with this conditional format.
+- `timeframe` (String) Defines the displayed timeframe.
+
+
+<a id="nestedblock--widget--heatmap_definition--request--formula--limit"></a>
+### Nested Schema for `widget.heatmap_definition.request.formula.limit`
+
+Optional:
+
+- `count` (Number) The number of results to return
```
Comment (estherk15) [inline docs/resources/dashboard.md:9653]:
```suggestion
- `count` (Number) The number of results to return.
```

File: docs/resources/dashboard.md [hunk 4]
```diff
@@ -9387,6 +9611,59 @@ Optional:
...
+- `order` (String) The direction of the sort. Valid values are `asc`, `desc`.
+
+
+<a id="nestedblock--widget--heatmap_definition--request--formula--style"></a>
+### Nested Schema for `widget.heatmap_definition.request.formula.style`
+
+Optional:
+
+- `palette` (String) The color palette used to display the formula. A guide to the available color palettes can be found at https://docs.datadoghq.com/dashboards/guide/widget_colors
```
Comment (estherk15) [inline docs/resources/dashboard.md:9662]:
```suggestion
- `palette` (String) The color palette used to display the formula. A guide to the available color palettes can be found at https://docs.datadoghq.com/dashboards/guide/widget_colors.
```

File: docs/resources/dashboard.md [hunk 5]
```diff
@@ -9465,6 +9742,173 @@ Optional:
...
+- `apm_resource_stats_query` (Block List, Max: 1) The APM Resource Stats query using formulas and functions. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--apm_resource_stats_query))
+- `event_query` (Block List, Max: 1) A timeseries formula and functions events query. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--event_query))
+- `metric_query` (Block List, Max: 1) A timeseries formula and functions metrics query. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--metric_query))
+- `process_query` (Block List, Max: 1) The process query using formulas and functions. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--process_query))
+- `slo_query` (Block List, Max: 1) The slo query using formulas and functions. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--slo_query))
```
Comment (estherk15) [inline docs/resources/dashboard.md:9755]:
```suggestion
- `slo_query` (Block List, Max: 1) The SLO query using formulas and functions. (see [below for nested schema](#nestedblock--widget--heatmap_definition--request--query--slo_query))
```

File: docs/resources/dashboard.md [hunk 6]
```diff
@@ -9465,6 +9742,173 @@ Optional:
...
+
+<a id="nestedblock--widget--heatmap_definition--request--query--apm_dependency_stats_query"></a>
+### Nested Schema for `widget.heatmap_definition.request.query.apm_dependency_stats_query`
+
+Required:
+
+- `data_source` (String) The data source for APM Dependency Stats queries. Valid values are `apm_dependency_stats`.
+- `env` (String) APM Environment.
```
Comment (estherk15) [inline docs/resources/dashboard.md:9763]:
```suggestion
- `env` (String) APM environment.
```

File: docs/resources/dashboard.md [hunk 7]
```diff
@@ -9465,6 +9742,173 @@ Optional:
...
+- `primary_tag_value` (String) Filter APM data by the second primary tag. `primary_tag_name` must also be specified.
+
+
+<a id="nestedblock--widget--heatmap_definition--request--query--apm_resource_stats_query"></a>
+### Nested Schema for `widget.heatmap_definition.request.query.apm_resource_stats_query`
+
+Required:
+
+- `data_source` (String) The data source for APM Resource Stats queries. Valid values are `apm_resource_stats`.
+- `env` (String) APM Environment.
```
Comment (estherk15) [inline docs/resources/dashboard.md:9783]:
```suggestion
- `env` (String) APM environment.
```

File: docs/resources/dashboard.md [hunk 8]
```diff
@@ -9465,6 +9742,173 @@ Optional:
...
+- `tag_filters` (List of String) An array of tags to filter by.
+- `text_filter` (String) The text to use as a filter.
+
+
+<a id="nestedblock--widget--heatmap_definition--request--query--slo_query"></a>
+### Nested Schema for `widget.heatmap_definition.request.query.slo_query`
+
+Required:
+
+- `data_source` (String) The data source for slo queries. Valid values are `slo`.
```
Comment (estherk15) [inline docs/resources/dashboard.md:9899]:
```suggestion
- `data_source` (String) The data source for SLO queries. Valid values are `slo`.
```

File: docs/resources/dashboard.md [hunk 9]
```diff
@@ -3798,6 +3853,173 @@ Optional:
...
+- `tag_filters` (List of String) An array of tags to filter by.
+- `text_filter` (String) The text to use as a filter.
+
+
+<a id="nestedblock--widget--group_definition--widget--heatmap_definition--request--query--slo_query"></a>
+### Nested Schema for `widget.group_definition.widget.heatmap_definition.request.query.slo_query`
+
+Required:
+
+- `data_source` (String) The data source for slo queries. Valid values are `slo`.
```
Comment (estherk15) [inline docs/resources/dashboard.md:4010]:
```suggestion
- `data_source` (String) The data source for SLO queries. Valid values are `slo`.
```

**Review by estherk15 — DISMISSED** [review]

**Review by estherk15 — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**